docs_user/assemblies/assembly_prepare_director_operator_and_rhoso_for_adoption_process.adoc

```diff
@@ -0,0 +1,190 @@
+
+[NOTE]
+====
+The example use case presented here is of a single OCP cluster, being shared by the source ODPdO env. _and_ a destination RHOSO env. Two OCP master nodes would be transferred to RHOSO use, leaving OSPdO with one for the adoption process, after which the remaining one can be used by RHOSO env. as well.
+====
+
+Work in the openstack project:
+[,bash]
+----
+oc project openstack
+----
+
+Define common environment variables
+[,bash]
+----
+PASSWORD_FILE="tripleo-passwords.yaml"
+export PASSWORD_FILE
+
+OSP18_NAMESPACE="osp18"
+export OSP18_NAMESPACE
+
+OSPDO_NAMESPACE="openstack"
+export OSPDO_NAMESPACE
+
+OS_CLIENT="oc exec -c openstackclient -t openstackclient -- "
+export OS_CLIENT
+
+CONTROLLER_SSH="oc rsh -n $OSPDO_NAMESPACE -c openstackclient openstackclient ssh controller-0.ctlplane"
+export CONTROLLER_SSH
+
+CONTROLLER1_SSH="oc -n $OSPDO_NAMESPACE rsh -c openstackclient openstackclient ssh controller-0.ctlplane"
+export CONTROLLER1_SSH
+CONTROLLER2_SSH=
+export CONTROLLER2_SSH
+CONTROLLER3_SSH=
+export CONTROLLER3_SSH
+
+OSPDO_INTERNAL_API_NET="internalapi"
+export OSPDO_INTERNAL_API_NET
+
+STORAGE_CLASS=$(oc get -n $OSPDO_NAMESPACE pvc openstackclient-hosts -o jsonpath='{.spec.storageClassName}')
+export STORAGE_CLASS
+----
+
+Get the node name upon which the current OSP controller VM is running.  This step will need to be modified with 3 running OSP controllers.  Modify to select one of the OSP nodes.
+[,bash]
+----
+CONTROLLER_NODE=$(oc -n $OSPDO_NAMESPACE get vmi -ojson | jq -r '.items[0].status.nodeName')
+export CONTROLLER_NODE
+----
+
+
+In case we're using a combo master/worker cluster with an HA OSPdO env,
+configure OSPdO to use a single Osp controller:
+
+Disable stonith:
+[,bash]
+----
+stonith_resources=$(sudo pcs stonith status|grep Started|awk '{print $2}')
+for in in stonith_resources ;do CONTROLLER1_SSH sudo pcs stonith disable ;done
+----
+Put non controller nodes in maintannace
+[,bash]
+----
+CONTROLLER1_SSH sudo pcs node maintenance controller-1 controller-2
+----
+Update quorum for single node:
+[,bash]
+----
+CONTROLLER1_SSH sudo systemctl stop corosync
+CONTROLLER2_SSH sudo systemctl stop corosync
+CONTROLLER3_SSH sudo systemctl stop corosync
+CONTROLLER1_SSH sudo pcs quorum update last_man_standing=1 wait_for_all=1
+CONTROLLER1_SSH sudo systemctl restart corosync
+----
+Restart pcmk cluster
+[,bash]
+----
+CONTROLLER1_SSH sudo pcs cluster stop
+CONTROLLER1_SSH sudo pcs cluster start
+----
+Remove OSPdO NNCPs from the other two nodes which are not running the controller VM by placing a nodeSelector on the OCP node containing the controller VM.
+[,bash]
+----
+for i in br-ctlplane br-ex br-osp; do oc patch osnetconfig openstacknetconfig --type json -p '[{"op": "replace", "path": "/spec/attachConfigurations/'$i'/nodeNetworkConfigurationPolicy/nodeSelector", "value": {"kubernetes.io/hostname": "'$CONTROLLER_NODE'"}}]'; done
+----
+
+== Preparing RHOSO for adoption
```

Comment:
This assembly is going to need to be split into 3 files:
- Top-level assembly
   -  Preparing director operator procedure
   -  Preparing RHOSO procedure
   
   I can do that after all the content is merged.

Reply:
ok sound good